aas_core_meta/v3.py

```diff
@@ -3461,7 +3461,8 @@ def __init__(
             )
         )
     ) or (
-            self.global_asset_ID is None
+            self.entity_type != Entity_type.Self_managed_entity
```

Comment:
@zrgt You actually don't need this check.

The first expression in the first operand is `self.entity_type == Entity_type.Self_managed_entity`. This first expression is then followed by an `or`. Hence, when the entity type is not self-managed entity, the first expression is invalid, and the truthness of the condition depends solely on the second part.

Please double-check. In case the readers were confused, we can gladly add this.

Reply:
Actually, I think it is possible to have the `and` part (3453-3463) to be `False` and the  `self.entity_type == Entity_type.Self_managed_entity` to be `True`, resulting in the whole first statement to be `False`, even though ` self.entity_type == Entity_type.Self_managed_entity`. 
Therefore it is necessary to check ` self.entity_type != Entity_type.Self_managed_entity` in the second statement after the `or`.

Reply:
@mristin try to run the invariant with the following case: 
`self.entity_type = Entity_type.Self_managed_entity`,
`self.global_asset_ID = None`
`self.specific_asset_ID = None`

The invariant without this additional check will return `True`, though the constraint forbid that for this case